### PR TITLE
Switch back to Downloads.jl

### PR DESCRIPTION
**Pull request by Wimmerer**
Sounds like this should be ready? Should I add a callback for progress?

### CODE REVIEW

File: src/ArchiveUtils.jl [hunk 1]
```diff
@@ -133,18 +133,7 @@ function download_verify(url, hash, path)
         @info "Cached file found in $(path)"
         return true
     else
-        mkpath(dirname(path))
```
Comment (giordano) [inline src/ArchiveUtils.jl:136]: I think the mkpath is still necessary

File: src/ArchiveUtils.jl [hunk 2]
```diff
@@ -133,18 +133,7 @@ function download_verify(url, hash, path)
...
-        # `curl` doesn't automatically overwrite an existing file, delete it
-        # before attempting the download.
```
Comment (giordano) [inline src/ArchiveUtils.jl:138]: We also want to keep the `@info` message and probably also the `rm` (I don't remember whether `Downloads.download` does it automatically)

**Comment by Wimmerer** [comment]
Had to change test file, but I believe that's correct. Do you want me to try and add a progress callback?

**Comment by giordano** [comment]
Thank you so much!

I don't have a strong opinion about progress bars, but in any case it needs to be optional, I don't think we want to on Yggdrasil for example, it'd make the log messy. In any case that's something for a different pull request

**Pull request merged**